bindings/python/iota_sdk/types/address.py

```diff
@@ -59,6 +63,57 @@ class NFTAddress:
     type: int = field(default_factory=lambda: int(AddressType.NFT), init=False)
 
 
+@json
+@dataclass
+class ImplicitAccountCreationAddress:
+    """Represents an implicit account creation address that can be used to transition an account.
+    Attributes:
+        address: The hex encoded Ed25519 Address.
+    """
+    address: Ed25519Address
+    type: int = field(default_factory=lambda: int(
+        AddressType.IMPLICIT_ACCOUNT_CREATION), init=False)
+
+    @staticmethod
+    def to_dict_custom(addr_dict: dict) -> dict:
```

Comment:
For my Python ignorance, why does `to_dict` takes a dict?

Reply:
no clue

Reply:
It's a static method, so doesn't use the values inside of the class
But actually it should be named `_to_dict_custom` to be used
https://github.com/iotaledger/iota-sdk/blob/346b6c897596607f9564483956ab6b7393fee04e/bindings/python/iota_sdk/types/common.py#L25-L29
And yeah looks a bit stupid, not sure why it's not a class method and just uses cls as parameter
Opened an issue https://github.com/iotaledger/iota-sdk/issues/1473